### PR TITLE
Improve JVM Arch Detection

### PR DESCRIPTION
**Pull request by smflorentino**
PR for #12256 Notify the user if the JVM architecture could not be detected, instead of silently defaulting to 32-bit. Also add a max heap parameter. On machines with RAM sizes in the order of 64GB plus, the initial heap could be multi-gigabyte. This can lead to an out of memory error when checking for the version.

### CODE REVIEW

**Comment by smflorentino** [comment]
I just signed the CLA, perhaps I need to recreate the PR?

File: core/bin/service.bat [hunk 1]
```diff
@@ -13,7 +13,12 @@ if not exist "%JAVA_HOME%\bin\java.exe" (
...
 )
-"%JAVA_HOME%\bin\java" -version 2>&1 | "%windir%\System32\find" "64-Bit" >nul:
+"%JAVA_HOME%\bin\java" -Xmx64M -version 2>&1 | "%windir%\System32\find" "64-Bit" >nul:
+
+if errorlevel 1 (
+	echo Warning: Unable to detect Java version, defaulting to x86  
+	goto x86
+)
 
 if errorlevel 1 goto x86
```
Comment (jpountz) [inline core/bin/service.bat]: is this line still needed then?

**Comment by jpountz** [comment]
Did you sign the CLA with a different name, I can't find your name in the list?

Regarding your PR, I know very little about windows but it looks like find will return an error level of 1 whenever it can't find the provided string. So this error message would be displayed for every person who is not running a 64-bits system. Doesn't it feel wrong?

**Comment by smflorentino** [comment]
@jpountz You're right. I forgot one small change which I've just pushed to the branch and squashed. I signed the CLA yesterday, and signed it again this morning, with my name/username...even got the email confirmation with the document. Should I forward you a copy?

**Comment by clintongormley** [comment]
@smflorentino did you sign the personal CLA or the corp CLA? We can only check the personal CLAs automatically, the corp ones we have to do a manual search on :(

**Comment by smflorentino** [comment]
The corp CLA. Should I also sign the personal one?

**Comment by clintongormley** [comment]
no worries @smflorentino - we've found the cla, thanks

**Comment by Mpdreamz** [comment]
LGTM :+1: small nag change from `-Xmx64M` to `-Xmx2M` just to signify that 64M has nothing to do with `x64`

**Comment by smflorentino** [comment]
@Mpdreamz do you think 2M is enough? I wouldn't want to cause a VM initialization failure because the heap was set too small. Just to be safe, maybe 50M or 128M (stated as the Java RAM requirement per the JVM docs). Otherwise I can change to 2M

**Comment by Mpdreamz** [comment]
50 works too, I read in the java docs it had to be a multiple of 1024kb but maybe 50 is a safer assumption to work over different JVM's :+1

**Comment by smflorentino** [comment]
I set it to 50M and tested successfully. Per http://docs.oracle.com/javase/7/docs/technotes/guides/vm/gc-ergonomics.html I think using Xmx arguments in megabytes will ensure that the value, in bytes, is a multiple of 1024. Do you think we are good to merge?

**Pull request merged**